### PR TITLE
deps, `options: filename`, multipolygon, `.d.ts`,  browser only downgrade (easy fix?)

### LINKED ISSUE #68: When export to shapefile, problem of combined polygon
Hello everone,
i have a geojson data. i exported a shp file but when i open at QGIS.  there is only 1 multipolygon. How can i seperate on javascript side ?

### LINKED ISSUE #73: download() returns shapefile with missings features
Given a geojson with multiple features download zip returns zip with only one feature. I have made sure the geojson is the correct format. Is it even possible to return a shapefile with multiple features?

### LINKED ISSUE #74: Name headers are truncated
Hi,
I've a data set with name headers that are truncated to 8 chars length, however, the body data is displayed correctly. Is there any way to output more than 8 chars in headers, just like in body?
Thanks!

Cheers,
Maxi

### LINKED ISSUE #88: I have 4 features in my collection, when downloaded file it puts coordinates of all 4 features into 1. I need all 4 features not 1, can anyone help?

### PR DESCRIPTION
**Pull request by charlieforward9**
closes #68
closes #73 
closes #74
closes #84
closes #88 
closes #90
opens https://github.com/mapbox/shp-write/issues/97

I guess my Prettier reformatted all the code, so it is hard to see my exact changes to the codebase. I do not have the time to change it to only show the pure diff.

install with: 
```
npm i https://github.com/Agriculture-Intelligence/shp-write.git
```

### CODE REVIEW

File: src/zip.js [hunk 1]
```diff
@@ -1,38 +1,49 @@
...
+  });
 
-    var generateOptions = { compression:'STORE' };
+  var generateOptions = { type: "base64", compression: "STORE" };
 
-    if (!process.browser) {
-      generateOptions.type = 'nodebuffer';
-    }
+  // Our implementation will DEFINITELY be in the browser for now.
+  // if (!process.browser) {
```
Comment (charlieforward9) [inline src/zip.js:42]: Our use case is run in the browser, so I removed this snippet of code. I am looking for someone with a NodeJS use case to contribute the respective server side solution before this is approved. Thank you.

Comment (sheindel) [reply src/zip.js:42]: Hey Charlie, please check the PR I opened (https://github.com/Agriculture-Intelligence/shp-write/pull/8) to your `master` branch that deals with this issues and others we had by abstracting more of this to the `options` parameter. It also cleans up the typing a bit. Thanks

Comment (sheindel) [reply src/zip.js:42]: Actually I typed it as `Promise<any> | ReadableStream;` and that works well enough for my use case for now while adding the `ReadableStream` for you.

Comment (charlieforward9) [reply src/zip.js:42]: I actually just added the stream bit as a copy/paste of [this contribution](https://github.com/mapbox/shp-write/pull/84). However, as I am working purely in the browser. I do not think I need that. Do with it as you please.

**Comment by charlieforward9** [comment]
@sheindel 
> [If we can make these changes I feel more confident pulling your PR into the main shp-write repo. Mapbox let me take it over](https://github.com/Agriculture-Intelligence/shp-write/pull/8#issuecomment-1684075715)

Seems like a good time to wrap this one up too!

**Pull request merged**